chia/util/struct_stream.py

```diff
@@ -78,13 +78,19 @@ def parse(cls: Type[_T_StructStream], f: BinaryIO) -> _T_StructStream:
         return cls.from_bytes(read_bytes)
 
     def stream(self, f: BinaryIO) -> None:
-        f.write(bytes(self))
+        f.write(self.stream_to_bytes())
 
     @classmethod
     def from_bytes(cls: Type[_T_StructStream], blob: bytes) -> _T_StructStream:  # type: ignore[override]
         if len(blob) != cls.SIZE:
             raise ValueError(f"{cls.__name__}.from_bytes() requires {cls.SIZE} bytes but got: {len(blob)}")
         return cls(int.from_bytes(blob, "big", signed=cls.SIGNED))
 
-    def __bytes__(self) -> bytes:
+    def stream_to_bytes(self) -> bytes:
         return super().to_bytes(length=self.SIZE, byteorder="big", signed=self.SIGNED)
+
+    # this is meant to avoid mixing up construcing a bytes object of a specific
+    # size (i.e. bytes(int)) vs. serializing the integer to bytes (i.e. bytes(uint32))
+    def __bytes__(self) -> bytes:
+        assert False
+        return b""
```

Comment:
https://docs.python.org/3.11/library/exceptions.html#NotImplementedError
> It should not be used to indicate that an operator or method is not meant to be supported at all – in that case either leave the operator / method undefined or, if a subclass, set it to [None](https://docs.python.org/3.11/library/constants.html#None).

So maybe we are supposed to do this?  I'm not really sure, haven't gone down this path much before.

```suggestion
    __bytes__ = None
```

Reply:
I'm a bit ignorant of when definitions end up in the class object and the instance object. I think methods always end up in the class object, but does `__bytes__ = None` also end up in the class object? It seems like it could also become a member of the instance object.

Reply:
It would be a class attribute and would also be accessible via instances since they fall back on the class if there's no instance attribute.

Given their recommendation, I was expecting `bytes()` do given a sensible error rather than just a generic type error.

```pycon
>>> class C(int):
...   __bytes__ = None
... 
>>> bytes(C())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'NoneType' object is not callable
```

I think something other than assert would be good though, and to provide a description and the intended replacement.

Reply:
https://mypy-play.net/?mypy=latest&python=3.11&gist=fff4a0a0cdbe9b7fdc4e00174443a5c5
```python
class C(int):
    __bytes__ = None

bytes(C())
```

This gives no error, so that isn't great.  Would be nice to find a way to catch this with mypy instead of at runtime.